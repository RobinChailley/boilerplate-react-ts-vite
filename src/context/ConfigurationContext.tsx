import { ConfigContextValueType } from '@context/configurationContextTypes';
import getConfigContextValue from '@context/getConfigurationContextValue';
import { createContext, useContext } from 'react';

export const ConfigContext = createContext<ConfigContextValueType>(getConfigContextValue());

export const useConfig = () => useContext(ConfigContext);
