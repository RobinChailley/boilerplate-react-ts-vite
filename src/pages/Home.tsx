import { useConfig } from '@context/ConfigurationContext';
import React from 'react';

interface HomeProps {

}

const Home: React.FC<HomeProps> = () => {
    const { translatorAdapter } = useConfig();
    const t = translatorAdapter.translate();

    return (
        <div>
            <h1 className={'text-3xl font-bold underline'}>{t('home.title')}</h1>
        </div>
    );
};

export default Home;
