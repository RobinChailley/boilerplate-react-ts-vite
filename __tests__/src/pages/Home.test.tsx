import { ConfigContext } from '@context/ConfigurationContext';
import Home from '@pages/Home';
import renderer from 'react-test-renderer';

import getConfigContextValue from '../../mocks/context/getConfigurationContextValue';

describe('Home', () => {
    it('should render correctly', () => {
        const render = renderer.create(
            <ConfigContext.Provider value={getConfigContextValue()}>
                <Home/>
            </ConfigContext.Provider>,
        );

        const snapshot = render!.toJSON();
        expect(snapshot).toMatchSnapshot();
    });
});
