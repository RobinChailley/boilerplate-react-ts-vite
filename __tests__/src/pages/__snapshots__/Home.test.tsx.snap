// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<div>
  <h1
    className="text-3xl font-bold text-white"
  >
    dashboard.title
  </h1>
</div>
`;
